### PR TITLE
statistics: fix the fomula for checking outdated stats

### LINKED ISSUE #26282: Update the formula about judging stale statistics
## Enhancement
Update the formula to `ModifyCnt / LastAnalyzeCnt`.

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: close #26282 

### What is changed and how it works?

Currently, we use `(modify count/realtime row count)` to check if the stats are outdated. But the reasonable method to calculate the ratio should be `(modify count/row count when last analyze)`.

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: statistics/table.go [hunk 1]
```diff
@@ -262,7 +264,11 @@ var RatioOfPseudoEstimate = atomic.NewFloat64(0.7)
 
 // IsOutdated returns true if the table stats is outdated.
 func (t *Table) IsOutdated() bool {
-	if t.Count > 0 && float64(t.ModifyCount)/float64(t.Count) > RatioOfPseudoEstimate.Load() {
+	rowcount := t.GetColRowCount()
+	if rowcount < 0 {
+		rowcount = float64(t.Count)
+	}
+	if t.Count > 0 && float64(t.ModifyCount)/rowcount > RatioOfPseudoEstimate.Load() {
```
Comment (qw4990) [inline statistics/table.go:271]: `t.Count > 0` --> `rowcount > 0`

**Review by qw4990 — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]

**Pull request merged**